### PR TITLE
Issue 410: Improving unit test coverage for controller and upgrade.go files

### LINKED ISSUE #410: Improving unit test coverage for controller and upgrade.go files
### Description
Improve unit test coverage for the files
pravegacluster_controller.go 
upgrade.go 
### Importance
Nice to have
### Location
pravegacluster_controller_test.go 
upgrade_test.go 

### Suggestions for an improvement

Add unit tests

### PR DESCRIPTION
**Pull request by anishakj**
### Change log description

Added unit tests for  the files
pravegacluster_controller.go
upgrade.go

### Purpose of the change

 Fixes #410

### What the code does

Improve test coverage

### How to verify it

unit tests should pass

### CODE REVIEW

File: pkg/controller/pravegacluster/upgrade_test.go [hunk 1]
```diff
@@ -275,6 +279,359 @@ var _ = Describe("Pravega Cluster Version Sync", func() {
...
+			It("Error should be nil when the target version is equal to current version", func() {
+				foundPravega.Status.SetUpgradingConditionTrue("UpgradeController", "0")
+				foundPravega.Status.TargetVersion = "0.6.1"
+				foundPravega.Status.CurrentVersion = "0.6.1"
+				r.client.Update(context.TODO(), foundPravega)
+				err = r.syncClusterVersion(foundPravega)
+				Ω(err).Should(BeNil())
+			})
+			It("Error should be not nil when the target version is not equal to current version", func() {
+				foundPravega.Status.SetUpgradingConditionTrue("UpgradeBookeeper", "0")
```
Comment (Prabhaker24) [inline pkg/controller/pravegacluster/upgrade_test.go:310]: here the message should not be "**UpgradeBookeeper**"

File: pkg/controller/pravegacluster/upgrade_test.go [hunk 2]
```diff
@@ -275,6 +279,359 @@ var _ = Describe("Pravega Cluster Version Sync", func() {
...
+				foundPravega = &v1beta1.PravegaCluster{}
+				_ = client.Get(context.TODO(), req.NamespacedName, foundPravega)
+				foundPravega.Status.SetUpgradingConditionTrue("UpgradeController", "0")
+				r.client.Update(context.TODO(), foundPravega)
+				err = r.syncClusterVersion(foundPravega)
+			})
+			It("Error should be nil when the target version is Empty", func() {
+				Ω(err).Should(BeNil())
+			})
+			It("Error should be nil when the target version is equal to current version", func() {
```
Comment (Prabhaker24) [inline pkg/controller/pravegacluster/upgrade_test.go:301]: we are only checking err is not nill here maybe we can change the assertion here

File: pkg/controller/pravegacluster/upgrade_test.go [hunk 3]
```diff
@@ -275,6 +279,359 @@ var _ = Describe("Pravega Cluster Version Sync", func() {
...
+				r.client.Create(context.TODO(), testpod)
+				r.client.Create(context.TODO(), testpod1)
+				r.client.Get(context.TODO(), types.NamespacedName{Name: "test", Namespace: "default"}, testpod)
+				r.client.Get(context.TODO(), types.NamespacedName{Name: "test1", Namespace: "default"}, testpod1)
+				pod = append(pod, testpod)
+				result1, err = r.checkUpdatedPods(pod, "0.7.1")
+				pod1 = append(pod1, testpod1)
+				result2, err1 = r.checkUpdatedPods(pod1, "0.7.1")
+			})
+			It("It should return false", func() {
```
Comment (Prabhaker24) [inline pkg/controller/pravegacluster/upgrade_test.go:628]: can we have different assertion for result1 and result2

File: pkg/controller/pravegacluster/upgrade_test.go [hunk 4]
```diff
@@ -508,6 +865,52 @@ var _ = Describe("Pravega Cluster Version Sync", func() {
...
+						_ = r.client.Get(context.TODO(), types.NamespacedName{Name: sts.Name, Namespace: p.Namespace}, sts)
+						_ = r.client.Get(context.TODO(), types.NamespacedName{Name: p.Name, Namespace: p.Namespace}, foundPravega)
+						sts.Status.ReadyReplicas = 2
+						*sts.Spec.Replicas = 2
+						r.client.Update(context.TODO(), sts)
+						foundPravega.Spec.Version = "0.5.0"
+						r.client.Update(context.TODO(), foundPravega)
+						result = r.rollbackConditionFor07(foundPravega, sts)
+					})
+					It("Error should not be nil", func() {
```
Comment (Prabhaker24) [inline pkg/controller/pravegacluster/upgrade_test.go:910]: Assertion description is not correct here

File: pkg/controller/pravegacluster/upgrade_test.go [hunk 5]
```diff
@@ -275,6 +279,359 @@ var _ = Describe("Pravega Cluster Version Sync", func() {
...
+				sts.Status.UpdatedReplicas = 5
+				sts.Status.Replicas = 3
+				r.client.Update(context.TODO(), sts)
+				r.client.Update(context.TODO(), foundPravega)
+				_, err1 = r.syncSegmentStoreVersion(foundPravega)
+			})
+			It("Error should be nil", func() {
+				Ω(err).ShouldNot(BeNil())
+			})
+			It("Error should be nil", func() {
```
Comment (Prabhaker24) [inline pkg/controller/pravegacluster/upgrade_test.go:440]: Assertion description is not correct here.

File: pkg/controller/pravegacluster/upgrade_test.go (hunk 1 above)
Comment (anishakj) [reply pkg/controller/pravegacluster/upgrade_test.go]: done

File: pkg/controller/pravegacluster/upgrade_test.go (hunk 3 above)
Comment (anishakj) [reply pkg/controller/pravegacluster/upgrade_test.go]: done

File: pkg/controller/pravegacluster/upgrade_test.go (hunk 4 above)
Comment (anishakj) [reply pkg/controller/pravegacluster/upgrade_test.go]: fixed

File: pkg/controller/pravegacluster/upgrade_test.go (hunk 5 above)
Comment (anishakj) [reply pkg/controller/pravegacluster/upgrade_test.go]: fixed

**Pull request merged**